cmd/syft/cli/attest/attest.go

```diff
@@ -183,23 +209,72 @@ func generateAttestation(predicate []byte, src *source.Source, sv *sign.SignerVe
 		return errors.Wrap(err, "unable to sign SBOM")
 	}
 
+	if app.Attest.NoUpload {
+		bus.Publish(partybus.Event{
+			Type: event.Exit,
+			Value: func() error {
+				_, err := os.Stdout.Write(signedPayload)
+				return err
+			},
+		})
+	}
+
+	return uploadAttestation(app, signedPayload, digest, sv)
+}
+
+func uploadAttestation(app *config.Application, signedPayload []byte, digest name.Digest, sv *sign.SignerVerifier) error {
```

Comment:
a great bonus here would be to add comments either in a block before hand or strewn throughout this function as to what is going on, noting the interaction with external entities and what gets stored where (what the transparency log is and what gets uploaded, what gets uploaded to the OCI registry and the important annotations, etc).

Reply:
comments added @wagoodman